File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        int n = digits.length;
        int[] ans = new int[n+1];
        for(int i=n-1; i>=0; i--) {
            if(carry == 0) {
                ans[i + 1] = digits[i];
                continue;
            }
            
            ans[i + 1] = digits[i] + carry;
            carry = 0;
            if(ans[i + 1] > 9) {
                carry = ans[i + 1]/10;
                ans[i + 1] %= 10;
            }
        }

        if(carry == 0) {
            return Arrays.copyOfRange(ans, 1, n+1);
        }

        ans[0] = carry;
        return ans;
    }
}

File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1

        for idx in range(len(digits)-1, -1, -1):
            new_digit = carry + digits[idx]
            if new_digit == 10:
                carry = 1
                digits[idx] = 0
            else:
                digits[idx] = new_digit
                return digits
        
        if carry == 1:
            digits.insert(0,1)
        
        return digits